contracts/src/ModuleCRC20Proxy.sol

```diff
@@ -0,0 +1,107 @@
+pragma solidity ^0.6.8;
+
+import "ds-math/math.sol";
+import "./ModuleCRC20.sol";
+
+contract ProxyAuthority {
+    address proxyAddress;
+
+    constructor(address _proxyAddress) public {
+        proxyAddress = _proxyAddress;
+    }
+
+    function canCall(
+        address src, address dst, bytes4 sig
+    ) public view returns (bool) {
+        if (msg.sender == proxyAddress) {
+            return true;
+        }
+
+        return false;
+    }
+}
+
+contract ModuleCRC20Proxy is DSMath {
+    // sha256('cronos-evm')[:20]
+    address constant module_address = 0x89A7EF2F08B1c018D5Cc88836249b84Dd5392905;
+    ModuleCRC20 crc20Contract;
+    bool isSource;
+
+    event __CronosSendToEvmChain(address indexed sender, address indexed recipient, uint256 indexed chain_id, uint256 amount, uint256 bridge_fee, bytes extraData);
+    event __CronosCancelSendToEvmChain(address indexed sender, uint256 id);
+
+    /**
+        Can be instantiated only by crc20 contract owner
+    **/
+    constructor(address crc20Contract_, bool isSource_) public {
+        crc20Contract = ModuleCRC20(crc20Contract_);
+        crc20Contract.setAuthority(DSAuthority(address(new ProxyAuthority(address(this)))));
+        isSource = isSource_;
+    }
+
+    /**
+        views
+    **/
+    function crc20() public view returns (address) {
+        return address(crc20Contract);
+    }
+
+    function is_source() public view returns (bool) {
+        return isSource;
+    }
+
+
+    /**
+        Internal functions to be called by cronos module.
+    **/
+    function mint_by_cronos_module(address addr, uint amount) public {
+        require(msg.sender == module_address);
+        crc20Contract.mint(addr, amount);
+    }
+
+    function burn_by_cronos_module(address addr, uint amount) public {
+        require(msg.sender == module_address);
+        crc20_burn(addr, amount);
+    }
+
+    function transfer_by_cronos_module(address addr, uint amount) public {
+        require(msg.sender == module_address);
+        crc20Contract.move(addr, module_address, amount);
+    }
+
+    function transfer_from_cronos_module(address addr, uint amount) public {
+        require(msg.sender == module_address);
```

Comment:
do we miss sender check in previous [ModuleCRC21](https://github.com/crypto-org-chain/cronos/blob/thomas/add-proxy-contract-example/contracts/src/ModuleCRC21.sol#L51)

Reply:
no but actually this checked is not needed because `transferFrom` already check that the src == msg.sender

I have added it here to make clear for the reader (but could probably be removed if we want to optimise the performance)